src/server/enterprise/cmds/cmds.go

```diff
@@ -192,6 +201,50 @@ func GetStateCmd() *cobra.Command {
 	return cmdutil.CreateAlias(getState, "enterprise get-state")
 }
 
+func SyncContextsCmd() *cobra.Command {
+	syncContexts := &cobra.Command{
+		Short: "Pull all available Pachyderm Cluster contexts into your pachctl config",
+		Long:  "Pull all available Pachyderm Cluster contexts into your pachctl config",
+		Run: cmdutil.Run(func(args []string) error {
+			cfg, err := config.Read(false, false)
+			if err != nil {
+				return err
+			}
+
+			c, err := client.NewOnUserMachine("user")
+			if err != nil {
+				return errors.Wrapf(err, "could not connect")
+			}
+			defer c.Close()
+			resp, err := c.License.ListUserClusters(c.Ctx(), &license.ListUserClustersRequest{})
+			if err != nil {
+				return err
+			}
+
+			// update the pach_address of all existing contexts, and add the rest as well.
+			for _, cluster := range resp.Clusters {
+				if context, ok := cfg.V2.Contexts[cluster.Id]; ok {
+					context.ClusterDeploymentID = cluster.ClusterDeploymentId
+					context.PachdAddress = cluster.Address
+				} else {
+					cfg.V2.Contexts[cluster.Id] = &config.Context{
+						ClusterDeploymentID: cluster.ClusterDeploymentId,
+						PachdAddress:        cluster.Address,
+						Source:              3, // what's the source? should we populate it here?
```

Comment:
I don't really know if we use the source for anything, but we could add a new value for "synced from the enterprise server" just in case.

Reply:
right now a Source value of 3 represents Imported. It makes sense to stick with that, right?

https://github.com/pachyderm/pachyderm/blob/master/src/internal/config/config.proto#L108